### ISSUE TITLE
Axom passing openmp flag to cuda when using axom-targets.cmake

### ISSUE THREAD

**Issue by corbett5** [LLNL/axom#105]
I just upgraded GEOSX to the 0.3.1 release and on Lassen the `axom-targets.cmake` includes the following lines
```cmake
set_target_properties(axom PROPERTIES
  INTERFACE_COMPILE_OPTIONS "-fopenmp=libomp"
  .
  .
  .
)
```

cmake winds up passing this flag to nvcc which then dies. If I comment out the `INTERFACE_COMPILER_OPTIONS line everything works.

Note: we are not building Axom with CUDA.

**Comment by gzagaris** [comment]
This is due to the way BLT currently handles compile definitions of an associated "target". Currently, no distinction is made between `PUBLIC` and `PRIVATE` and the default behavior is for them to be `PUBLIC`, which propagates these flags to other targets using Axom. 

See related BLT issues:

1. https://github.com/LLNL/blt/issues/307
2. https://github.com/LLNL/blt/issues/305

**Comment by cyrush** [comment]
This is a tricky one. Hard for Axom to know to setup the flags correctly for the cuda case when it doesn't build with cuda. Not sure it can export things correctly without knowing the downstream linking language. 

Also, if there is openmp usage in the headers of Axom, you will need those flags.

**Comment by gzagaris** [comment]
Yeah, it gets tricky. However, I think if we make things like openmp `PRIVATE`, the calling application can  always explicitly add it if needed. Seems like it would always be easier to add things at the application level, than remove flags that are added under the covers.

**Comment by kennyweiss** [comment]
@rrsettgast -- at your convenience, please check out #225 which improves axom's export.

Notably, I am clearing Axom's and Raja's ``INTERFACE_COMPILER_OPTIONS`` flag in OpenMP configurations to avoid exporting ``openmp`` flags.  This is a stopgap until Axom or BLT improves its handling of OpenMP flags.

**Comment by kennyweiss** [comment]
Also:
* Issue #239 relates to testing Axom's CUDA-based installations
* Issue #118 will add an OpenMP + CUDA host-config on LLNL's `blueos`, which will likely shake out some additional problems.

**Comment by rhornung67** [comment]
@kennyweiss @white238 is this issue still relevant? if not, can we close it?
